python/lsst/ts/hexrotcomm/base_csc.py

```diff
@@ -311,26 +312,47 @@ def assert_summary_state(self, *allowed_states, isbefore):
 
         Parameters
         ----------
-        allowed_states : `List` [`lsst.ts.salobj.State`]
-            Allowed summary states.
-        isbefore : `bool`
-            Determines the error message prefix.
-            Set True if checking initial conditions before commanding
-            the low-level controller. Message is "Rejected: initial state ...".
-            Set False if checking the state after commanding
-            the low-level controller. Message is "Failed: final state ..."
+        state : `lsst.ts.salobj.State`
+            Expected summary state.
+        isbefore : `bool`, optional
+            Deprecated. The only allowed values are False
+            (which raises a deprecation warning) and None.
         """
+        if isbefore:
+            raise ValueError(
+                f"isbefore={isbefore}; this deprecated argument must be None or False"
+            )
+        elif isbefore is False:
+            warnings.warn(f"isbefore={isbefore} is deprecated", DeprecationWarning)
+        state = salobj.State(state)
         self.assert_commandable()
-        if self.summary_state not in allowed_states:
-            allowed_states_str = ", ".join(repr(state) for state in allowed_states)
-            if isbefore:
-                msg_prefix = "Rejected: initial"
-            else:
-                msg_prefix = "Failed: final"
+        if self.summary_state != state:
             raise salobj.ExpectedError(
-                f"{msg_prefix} state is {self.summary_state!r} instead of {allowed_states_str}"
+                f"Rejected: initial state is {self.summary_state!r} instead of {state!r}"
             )
 
+    async def wait_summary_state(self, state, max_telem=5):
```

Comment:
Consider making `max_telem` a startup configuration parameter (one that we could control when we start the CSC). I worry we may have to tweak this number a bit and the way it is implemented it would require a code change.

Reply:
Unfortunately this parameter is required before the CSC goes into the "start" state. So the best I could do is start with a default value that the "start" command overrides. That's complicated and difficult to explain to users. So would much rather start out with a constant to see if it suffices. I chose 5 to be quite generous.

Reply:
However, I did make it a named constant, so it's easier to find.

Reply:
If telemetry is 20Hz, 5 means 0.25 seconds right? I have the impression it might still be too short. But, we will test it tomorrow and I'll let you know..

Reply:
Since it passed the tests we'll leave it at 5 for now. I'll gladly increase it to 10 if we find we need more.